contracts/implementation/PoolCommitter.sol

```diff
@@ -59,301 +48,247 @@ contract PoolCommitter is IPoolCommitter, Ownable {
+            shortBalance,
+            _commits.shortBurnAmount
+        );
+
+        uint256 newLongBalance = longBalance + _commits.longMintAmount - longBurnAmount;
+        uint256 newShortBalance = shortBalance + _commits.shortMintAmount - shortBurnAmount;
+
+        // TODO update user's aggregate balance
+        // TODO setNewPoolBalances once at end
+        // TODO only mint/burn if amount is > 0
+        updateIntervalId += 1;
+
+        // Update the collateral on each side
+        pool.setNewPoolBalances(newLongBalance, newShortBalance);
     }
 
-    function _uncommit(Commit memory _commit, uint128 _commitID) internal {
-        // reduce pool commitment amount
-        uint256 _commitType = uint256(_commit.commitType);
-        shadowPools[_commitType] = shadowPools[_commitType] - _commit.amount;
-        emit RemoveCommit(_commitID, _commit.amount, _commit.commitType);
+    function executeCommitments() external override onlyPool {
+        Commitment memory _commits = totalMostRecentCommit;
+        ILeveragedPool pool = ILeveragedPool(leveragedPool);
+        executeGivenCommitments(totalMostRecentCommit);
 
-        delete commits[_commitID];
+        totalMostRecentCommit = totalNextIntervalCommit;
+        delete totalNextIntervalCommit;
 
-        if (earliestCommitUnexecuted == _commitID) {
-            // This is the first unexecuted commit, so we can bump this up one
-            earliestCommitUnexecuted += 1;
-            skipDeletedMiddleCommits(ScanDirection.UP);
-        }
-        if (latestCommitUnexecuted == _commitID && earliestCommitUnexecuted != NO_COMMITS_REMAINING) {
-            // This is the latest commit unexecuted that we are trying to delete.
-            latestCommitUnexecuted -= 1;
-            skipDeletedMiddleCommits(ScanDirection.DOWN);
+        uint32 two = 2;
+        if (block.timestamp >= pool.lastPriceTimestamp() + pool.updateInterval() * two) {
+            // Another update interval has passed, so we have to do the nextIntervalCommit as well
+            executeGivenCommitments(totalMostRecentCommit);
+            delete totalMostRecentCommit;
         }
+    }
 
-        // release tokens
-        if (_commit.commitType == CommitType.LongMint || _commit.commitType == CommitType.ShortMint) {
-            // minting: return quote tokens to the commit owner
-            ILeveragedPool(leveragedPool).quoteTokenTransfer(msg.sender, _commit.amount);
-        } else if (_commit.commitType == CommitType.LongBurn) {
-            // long burning: return long pool tokens to commit owner
-            ILeveragedPool(leveragedPool).mintTokens(0, _commit.amount, msg.sender);
-        } else if (_commit.commitType == CommitType.ShortBurn) {
-            // short burning: return short pool tokens to the commit owner
-            ILeveragedPool(leveragedPool).mintTokens(1, _commit.amount, msg.sender);
-        }
+    function updateSingleCommitment(address user, Commitment memory commit)
```

Comment:
`user` isn't being used in this function (you're already passing the `Commitment` of the user)

Reply:
You should also rename `commit` to something else, compiler gives a warning about it